ouroboros-network/src/Ouroboros/Network/PeerSelection/Simple.hs

```diff
@@ -68,6 +71,17 @@ withPeerSelectionActions localRootTracer publicRootTracer timeout targets _stati
             (a :| as))
           (\thread -> k (Just thread) peerSelectionActions)
   where
+    -- We first try to get poublic root peers from the ledger, but if it fails
+    -- (for example because the node hasn't synced far enough) we fall back
+    -- to using the manually configured bootstrap root peers.
+    requestLedgerPeers :: Int -> IO (Set Socket.SockAddr, DiffTime)
+    requestLedgerPeers n = do
+        atomically $ reqLedgerPeers $ NumberOfPeers $ fromIntegral n
+        peers_m <- atomically getLedgerPeers
```

Comment:
If we make `getLedgerPeers` an `IO` action we can pass the request as its argument, and avoid exposing internal communication.

Reply:
done